src/components/search-result-list/search-result-list.js

```
@@ -0,0 +1,111 @@
+import React, { PropTypes } from 'react';
+import SearchResultItem from '../search-result-item';
+import SearchResultPagination from '../search-result-pagination';
+
+function render(data) {
+  const list = data.results.map((result, i) => {
+    let title;
+    let url;
+    let description;
+
+    if (typeof result !== 'undefined') {
+      title = result.title;
+      url = result.url;
+      description = result.description;
+    }
+
+    return (
+      <SearchResultItem
+        key={i}
+        title={title}
+        url={url}
+        description={description}
+      />);
+  });
+
+  return (
+    <div className="search-list">
+      <div className="search-list__list">
+        {list}
+      </div>
+      <div className="search-list__pagination">
+        <SearchResultPagination
+          pages={data.pages}
+          page={data.actualPage}
+          lastPage={data.lastPage}
+          paginate={data.paginate}
+        />
+      </div>
+    </div>
+  );
+}
+
+function SearchResultList(props) {
+  const results = props.results.json;
+
+  if (!Array.isArray(results)) {
```

What can it be? Why do we need this safeguard? do we want to at least ```console.warn``` something went wrong? My main worry is that we discard errors which shouldn't happen without noticing for a long time and making debugging of a problem way longer.
Also, if props.results is undefined we crash on line 44, so this condition only saves for existing results but not in array form